src/jupyterlabcontroller/storage/k8s.py

```diff
@@ -0,0 +1,518 @@
+    V1ObjectMeta,
+    V1Pod,
+    V1PodSecurityContext,
+    V1PodSpec,
+    V1ResourceFieldSelector,
+    V1ResourceQuota,
+    V1ResourceQuotaSpec,
+    V1Secret,
+    V1SecretVolumeSource,
+    V1SecurityContext,
+    V1Toleration,
+    V1Volume,
+    V1VolumeMount,
+)
+from kubernetes_asyncio.client.rest import ApiException
+from kubernetes_asyncio.watch import Watch
+from structlog.stdlib import BoundLogger
+
+from ..config import LabSecret
+from ..models.exceptions import (
+    KubernetesError,
+    NSCreationError,
+    WaitingForObjectError,
+    WatchError,
+)
+from ..models.k8s import (
+    ContainerImage,
+    NodeContainers,
+    ObjectOperation,
+    Secret,
+)
+from ..models.v1.event import Event, EventQueue
+from ..models.v1.lab import UserResourceQuantum
+
+# FIXME
+# For now these are just aliases, but we want to do what we did with
+# ContainerImage above and create simplified versions of the objects
+# with some defaults held constant.
+#
+# Not sure this is even worth it since K8s does a perfectly adequate job
+# of making models for all its types.  Maybe we should just use them
+# directly?  Not very many can be usefully simplified.
+Affinity: TypeAlias = V1Affinity
+ConfigMapEnvSource: TypeAlias = V1ConfigMapEnvSource
+ConfigMapVolumeSource: TypeAlias = V1ConfigMapVolumeSource
+Container: TypeAlias = V1Container
+DownwardAPIVolumeFile: TypeAlias = V1DownwardAPIVolumeFile
+DownwardAPIVolumeSource: TypeAlias = V1DownwardAPIVolumeSource
+EmptyDirVolumeSource: TypeAlias = V1EmptyDirVolumeSource
+EnvFromSource: TypeAlias = V1EnvFromSource
+HostPathVolumeSource: TypeAlias = V1HostPathVolumeSource
+KeyToPath: TypeAlias = V1KeyToPath
+LocalObjectReference: TypeAlias = V1LocalObjectReference
+NFSVolumeSource: TypeAlias = V1NFSVolumeSource
+NodeAffinity: TypeAlias = V1NodeAffinity
+NodeSelector: TypeAlias = V1NodeSelector
+NodeSelectorRequirement: TypeAlias = V1NodeSelectorRequirement
+NodeSelectorTerm: TypeAlias = V1NodeSelectorTerm
+ObjectFieldSelector: TypeAlias = V1ObjectFieldSelector
+PodSecurityContext: TypeAlias = V1PodSecurityContext
+PodSpec: TypeAlias = V1PodSpec
+ResourceFieldSelector: TypeAlias = V1ResourceFieldSelector
+SecretVolumeSource: TypeAlias = V1SecretVolumeSource
+SecurityContext: TypeAlias = V1SecurityContext
+Toleration: TypeAlias = V1Toleration
+Volume: TypeAlias = V1Volume
+VolumeMount: TypeAlias = V1VolumeMount
+
+# Weirdly, it turns out that NFS is internally an alias to AWS EBS.
+
+
+class K8sStorageClient:
+    def __init__(
+        self, k8s_api: ApiClient, timeout: int, logger: BoundLogger
+    ) -> None:
+        self.k8s_api = k8s_api
+        self.api = client.CoreV1Api(k8s_api)
+        self.timeout = timeout
+        self.logger = logger
+
+    async def aclose(self) -> None:
+        await self.k8s_api.close()
+
+    def get_std_metadata(self, name: str) -> V1ObjectMeta:
+        return V1ObjectMeta(
+            name=name,
+            labels={"argocd.argoproj.io/instance": "nublado-users"},
```

Comment:
I would make the namespace a constant in `jupyterlabcontroller.constants` rather than hard-coding it here. The extra annotations probably should be a constant too.